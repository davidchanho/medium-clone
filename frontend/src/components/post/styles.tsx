import styled from "styled-components";

export const Container = styled.div``;

export const Header = styled.header`
  display: flex;
  align-items: center;
`;

export const Icon = styled.img`
  border-radius: 50%;
`;
