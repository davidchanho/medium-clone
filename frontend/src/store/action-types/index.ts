export enum ActionTypes {
  FETCH_POSTS = "FETCH_POSTS",
  FETCH_POSTS_SUCCESS = "FETCH_POSTS_SUCCESS",
  FETCH_POSTS_FAIL = "FETCH_POSTS_FAIL",
  FETCH_POST = "FETCH_POST",
  FETCH_POST_SUCCESS = "FETCH_POST_SUCCESS",
  FETCH_POST_FAIL = "FETCH_POST_FAIL",
  ADD_POST = "ADD_POST",
  DELETE_POST = "DELETE_POST",
  UPDATE_POST = "UPDATE_POST",
}
