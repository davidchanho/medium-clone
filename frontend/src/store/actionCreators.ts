import { Dispatch } from "react";
import { Action, ActionTypes } from ".";
import { IComment, IPost, IUser } from "../types";
import API from "./api";

export const getComments = () => async (dispatch: Dispatch<Action>) => {
  dispatch({ type: ActionTypes.FETCH_COMMENTS });

  try {
    const data = await API.getComments();
    dispatch({
      type: ActionTypes.FETCH_COMMENTS_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_COMMENTS_FAIL,
      payload: err,
    });
  }
};

export const getComment = (_id: string) => async (
  dispatch: Dispatch<Action>
) => {
  dispatch({ type: ActionTypes.FETCH_COMMENT });

  try {
    const data = await API.getComment(_id);
    dispatch({
      type: ActionTypes.FETCH_COMMENT_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_COMMENT_FAIL,
      payload: err,
    });
  }
};

export const addComment = (comment: IComment) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    API.addComment(comment);
    dispatch({ type: ActionTypes.ADD_COMMENT, payload: comment });
  } catch (err) {
    console.log(err);
  }
};

export const deleteComment = (_id: string) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    API.deleteComment(_id);
    dispatch({
      type: ActionTypes.DELETE_COMMENT,
      payload: _id,
    });
  } catch (err) {
    console.log(err);
  }
};

export const updateComment = (comment: IComment) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    API.updateComment(comment);
    dispatch({
      type: ActionTypes.UPDATE_COMMENT,
      payload: comment,
    });
  } catch (err) {
    console.log(err);
  }
};

export const getPosts = () => async (dispatch: Dispatch<Action>) => {
  dispatch({ type: ActionTypes.FETCH_POSTS });
  try {
    const data = await API.getPosts();

    dispatch({
      type: ActionTypes.FETCH_POSTS_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_POSTS_FAIL,
      payload: err,
    });
  }
};

export const getPost = (_id: string) => async (dispatch: Dispatch<Action>) => {
  dispatch({ type: ActionTypes.FETCH_POST });

  try {
    const data = await API.getPost(_id);

    dispatch({
      type: ActionTypes.FETCH_POST_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_POST_FAIL,
      payload: err,
    });
  }
};

export const addPost = (post: IPost) => async (dispatch: Dispatch<Action>) => {
  try {
    API.addPost(post);

    dispatch({ type: ActionTypes.ADD_POST, payload: post });
  } catch (err) {
    console.log(err);
  }
};

export const dismissPost = (_id: string) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    dispatch({
      type: ActionTypes.DISMISS_POST,
      payload: _id,
    });
  } catch (err) {
    console.log(err);
  }
};

export const updatePost = (post: IPost) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    API.updatePost(post);

    dispatch({
      type: ActionTypes.UPDATE_POST,
      payload: post,
    });
  } catch (err) {
    console.log(err);
  }
};

export const getPublications = () => async (dispatch: Dispatch<Action>) => {
  dispatch({ type: ActionTypes.FETCH_PUBLICATIONS });

  try {
    const data = await API.getPublications();
    dispatch({
      type: ActionTypes.FETCH_PUBLICATIONS_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_PUBLICATIONS_FAIL,
      payload: err,
    });
  }
};

export const getPublication = (_id: string) => async (
  dispatch: Dispatch<Action>
) => {
  dispatch({ type: ActionTypes.FETCH_PUBLICATION });

  try {
    const data = await API.getPublication(_id);
    dispatch({
      type: ActionTypes.FETCH_PUBLICATION_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_PUBLICATION_FAIL,
      payload: err,
    });
  }
};

export const getUsers = () => async (dispatch: Dispatch<Action>) => {
  dispatch({ type: ActionTypes.FETCH_USERS });

  try {
    const data = await API.getUsers();
    dispatch({
      type: ActionTypes.FETCH_USERS_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_USERS_FAIL,
      payload: err,
    });
  }
};

export const getUser = (_id: string) => async (dispatch: Dispatch<Action>) => {
  dispatch({ type: ActionTypes.FETCH_USER });

  try {
    const data = await API.getUser(_id);
    dispatch({
      type: ActionTypes.FETCH_USER_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_USER_FAIL,
      payload: err,
    });
  }
};

export const addUser = (user: IUser) => async (dispatch: Dispatch<Action>) => {
  try {
    API.addUser(user);
    dispatch({ type: ActionTypes.ADD_USER, payload: user });
  } catch (err) {
    console.log(err);
  }
};

export const deleteUser = (_id: string) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    API.deleteUser(_id);
    dispatch({
      type: ActionTypes.DELETE_USER,
      payload: _id,
    });
  } catch (err) {
    console.log(err);
  }
};

export const updateUser = (user: IUser) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    API.updateUser(user);
    dispatch({
      type: ActionTypes.UPDATE_USER,
      payload: user,
    });
  } catch (err) {
    console.log(err);
  }
};

export const bookmarkPost = (post: IPost) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    dispatch({
      type: ActionTypes.BOOKMARK_POST,
      payload: post,
    });
  } catch (err) {
    console.log(err);
  }
};

export const getTopics = () => async (dispatch: Dispatch<Action>) => {
  dispatch({ type: ActionTypes.FETCH_TOPICS });

  try {
    const data = await API.getTopics();
    dispatch({
      type: ActionTypes.FETCH_TOPICS_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_TOPICS_FAIL,
      payload: err,
    });
  }
};
