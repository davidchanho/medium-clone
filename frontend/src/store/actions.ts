import { ActionTypes } from ".";
import { IComment, IItem, IPost, IPublication, ITopic, IUser } from "../types";

export interface LoadingItemsAction {
  type: ActionTypes.LOADING_ITEMS;
}

export interface ErrorItemsAction {
  type: ActionTypes.ERROR_ITEMS;
  name: string;
  payload: string;
}

export interface FetchItemsAction {
  type: ActionTypes.FETCH_ITEMS;
  name: string;
  payload: IItem[];
}

export interface FetchItemAction {
  type: ActionTypes.FETCH_ITEM;
  name: string;
  payload: IItem;
}

export interface AddItemAction {
  type: ActionTypes.ADD_ITEM;
  name: string;
  payload: IItem;
}

export interface DeleteItemAction {
  type: ActionTypes.DELETE_ITEM;
  name: string;
  payload: string;
}

export interface UpdateItemAction {
  type: ActionTypes.UPDATE_ITEM;
  name: string;
  payload: IItem;
}

export interface LoadingCommentsAction {
  type: ActionTypes.LOADING_COMMENTS;
}

export interface ErrorCommentsAction {
  type: ActionTypes.ERROR_COMMENTS;
  payload: string;
}

export interface FetchCommentsAction {
  type: ActionTypes.FETCH_COMMENTS;
  payload: IComment[];
}

export interface FetchCommentAction {
  type: ActionTypes.FETCH_COMMENT;
  payload: IComment;
}

export interface AddCommentAction {
  type: ActionTypes.ADD_COMMENT;
  payload: IComment;
}

export interface DeleteCommentAction {
  type: ActionTypes.DELETE_COMMENT;
  payload: string;
}

export interface UpdateCommentAction {
  type: ActionTypes.UPDATE_COMMENT;
  payload: IComment;
}

export interface LoadingPostsAction {
  type: ActionTypes.LOADING_POSTS;
}

export interface FetchPostsAction {
  type: ActionTypes.FETCH_POSTS;
  payload: IPost[];
}

export interface ErrorPostsAction {
  type: ActionTypes.ERROR_POSTS;
  payload: string;
}

export interface FetchPostAction {
  type: ActionTypes.FETCH_POST;
  payload: IPost;
}

export interface AddPostAction {
  type: ActionTypes.ADD_POST;
  payload: IPost;
}

export interface DeletePostAction {
  type: ActionTypes.DELETE_POST;
  payload: string;
}

export interface UpdatePostAction {
  type: ActionTypes.UPDATE_POST;
  payload: IPost;
}

export interface LoadingPublicationsAction {
  type: ActionTypes.LOADING_PUBLICATIONS;
}

export interface FetchPublicationsAction {
  type: ActionTypes.FETCH_PUBLICATIONS;
  payload: IPublication[];
}

export interface ErrorPublicationsAction {
  type: ActionTypes.ERROR_PUBLICATIONS;
  payload: string;
}

export interface FetchPublicationAction {
  type: ActionTypes.FETCH_PUBLICATION;
  payload: IPublication;
}

export interface LoadingTopicsAction {
  type: ActionTypes.LOADING_TOPICS;
}

export interface FetchTopicsAction {
  type: ActionTypes.FETCH_TOPICS;
  payload: ITopic[];
}

export interface ErrorTopicsAction {
  type: ActionTypes.ERROR_TOPICS;
  payload: string;
}

export interface LoadingUsersAction {
  type: ActionTypes.LOADING_USERS;
}

export interface FetchUsersAction {
  type: ActionTypes.FETCH_USERS;
  payload: IUser[];
}

export interface ErrorUsersAction {
  type: ActionTypes.ERROR_USERS;
  payload: string;
}

export interface FetchUserAction {
  type: ActionTypes.FETCH_USER;
  payload: IUser;
}

export interface AddUserAction {
  type: ActionTypes.ADD_USER;
  payload: IUser;
}

export interface DeleteUserAction {
  type: ActionTypes.DELETE_USER;
  payload: string;
}

export interface UpdateUserAction {
  type: ActionTypes.UPDATE_USER;
  payload: IUser;
}

export interface BookmarkPostAction {
  type: ActionTypes.BOOKMARK_POST;
  payload: IPost;
}

export type Action =
  | FetchItemsAction
  | ErrorItemsAction
  | LoadingItemsAction
  | FetchItemAction
  | AddItemAction
  | DeleteItemAction
  | UpdateItemAction
  | FetchCommentsAction
  | ErrorCommentsAction
  | LoadingCommentsAction
  | FetchCommentAction
  | AddCommentAction
  | DeleteCommentAction
  | UpdateCommentAction
  | FetchPostsAction
  | ErrorPostsAction
  | LoadingPostsAction
  | FetchPostAction
  | AddPostAction
  | DeletePostAction
  | UpdatePostAction
  | FetchPublicationsAction
  | ErrorPublicationsAction
  | LoadingPublicationsAction
  | FetchPublicationAction
  | FetchTopicsAction
  | ErrorTopicsAction
  | LoadingTopicsAction
  | FetchUsersAction
  | ErrorUsersAction
  | LoadingUsersAction
  | FetchUserAction
  | AddUserAction
  | DeleteUserAction
  | UpdateUserAction
  | BookmarkPostAction;
