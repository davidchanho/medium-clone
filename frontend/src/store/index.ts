export * as actionCreators from "./actionCreators";
export * from "./actions";
export * from "./actionTypes";
export * from "./api";
export * from "./reducers";
export * from "./selectors";
export * from "./store";
