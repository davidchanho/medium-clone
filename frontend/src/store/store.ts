import localForage from "localforage";
import { applyMiddleware, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import { createLogger } from "redux-logger";
import { persistReducer, persistStore } from "redux-persist";
import thunk from "redux-thunk";
import reducers from "./reducers";

// localForage
localForage.config({
  driver: localForage.INDEXEDDB,
});

// redux-perist
const persistConfig = {
  key: "medium",
  storage: localForage,
  whitelist: ["posts"],
};
const persistedReducer = persistReducer(persistConfig, reducers);

// middleware
const logger = createLogger();
const middleware = [thunk, logger];

export const store = createStore(
  persistedReducer,
  {},
  composeWithDevTools(applyMiddleware(...middleware))
);

export const persistor = persistStore(store);

export type RootState = ReturnType<typeof store.getState>;
