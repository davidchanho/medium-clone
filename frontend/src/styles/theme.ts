import * as breakpoints from "./breakpoints";
import * as colors from "./colors";

export const theme = () => ({
  breakpoints: {
    mobile: breakpoints.mobile,
    tablet: breakpoints.tablet,
    desktop: breakpoints.desktop,
  },
  colors: {
    primaryButton: colors.white["100"],
    primaryButtonOutline: colors.grey["100"],
    primaryButtonText: colors.grey["100"],

    secondaryText: colors.grey["100"],
    secondaryTextHover: colors.black["200"],

    numberedText: colors.grey["200"],

    successButton: colors.green["100"],
    successButtonHover: colors.green["200"],
    successButtonText: colors.white["100"],

    successText: colors.green["100"],

    darkBackground: colors.black["200"],
    darkText: colors.black["100"],

    white: colors.white["100"],
  },
});
