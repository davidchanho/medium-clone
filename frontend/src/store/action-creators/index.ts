import { Dispatch } from "redux";
import db from "../../api";
import { ActionTypes } from "../action-types";
import { Action } from "../actions";
import { IPost, IPublication } from "../reducers/posts";

export const fetchPosts = () => async (dispatch: Dispatch<Action>) => {
  dispatch({ type: ActionTypes.FETCH_POSTS });

  try {
    const data = await db.fetchPosts();
    dispatch({
      type: ActionTypes.FETCH_POSTS_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_POSTS_FAIL,
      payload: err,
    });
  }
};

export const fetchPost = (_id: string) => async (
  dispatch: Dispatch<Action>
) => {
  dispatch({ type: ActionTypes.FETCH_POST });

  try {
    const data = await db.fetchPost(_id);
    dispatch({
      type: ActionTypes.FETCH_POST_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_POST_FAIL,
      payload: err,
    });
  }
};

export const addPost = (post: IPost) => async (dispatch: Dispatch<Action>) => {
  try {
    db.addPost(post);
    dispatch({ type: ActionTypes.ADD_POST, payload: post });
  } catch (err) {
    console.log(err);
  }
};

export const deletePost = (_id: string) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    db.deletePost(_id);
    dispatch({
      type: ActionTypes.DELETE_POST,
      payload: _id,
    });
  } catch (err) {
    console.log(err);
  }
};

export const updatePost = (post: IPost) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    db.updatePost(post);
    dispatch({
      type: ActionTypes.UPDATE_POST,
      payload: post,
    });
  } catch (err) {
    console.log(err);
  }
};

export const fetchPublications = () => async (dispatch: Dispatch<Action>) => {
  dispatch({ type: ActionTypes.FETCH_PUBLICATIONS });

  try {
    const data = await db.fetchPublications();
    dispatch({
      type: ActionTypes.FETCH_PUBLICATIONS_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_PUBLICATIONS_FAIL,
      payload: err,
    });
  }
};

export const fetchPublication = (_id: string) => async (
  dispatch: Dispatch<Action>
) => {
  dispatch({ type: ActionTypes.FETCH_PUBLICATION });

  try {
    const data = await db.fetchPublication(_id);
    dispatch({
      type: ActionTypes.FETCH_PUBLICATION_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_PUBLICATION_FAIL,
      payload: err,
    });
  }
};

export const addPublication = (publication: IPublication) => async (dispatch: Dispatch<Action>) => {
  try {
    db.addPublication(publication);
    dispatch({ type: ActionTypes.ADD_PUBLICATION, payload: publication });
  } catch (err) {
    console.log(err);
  }
};

export const deletePublication = (_id: string) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    db.deletePublication(_id);
    dispatch({
      type: ActionTypes.DELETE_PUBLICATION,
      payload: _id,
    });
  } catch (err) {
    console.log(err);
  }
};

export const updatePublication = (publication: IPublication) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    db.updatePublication(publication);
    dispatch({
      type: ActionTypes.UPDATE_PUBLICATION,
      payload: publication,
    });
  } catch (err) {
    console.log(err);
  }
};
