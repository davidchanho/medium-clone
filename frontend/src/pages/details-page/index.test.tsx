import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import PostDetailsPage from ".";
import { store } from "../../store";
import { theme } from "../../styles/theme";

describe("PostDetailsPage", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ThemeProvider theme={theme}>
            <Router>
              <PostDetailsPage />
            </Router>
          </ThemeProvider>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
