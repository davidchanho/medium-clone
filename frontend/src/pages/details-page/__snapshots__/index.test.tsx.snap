// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostDetailsPage renders correctly 1`] = `
<div>
  <div>
    <div>
      <p>
        
      </p>
      <button>
        Follow
      </button>
    </div>
     
    <div />
    <div>
      <div>
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M372.57 112.641v-10.825c0-43.612-40.52-76.691-83.039-65.546-25.629-49.5-94.09-47.45-117.982.747C130.269 26.456 89.144 57.945 89.144 102v126.13c-19.953-7.427-43.308-5.068-62.083 8.871-29.355 21.796-35.794 63.333-14.55 93.153L132.48 498.569a32 32 0 0 0 26.062 13.432h222.897c14.904 0 27.835-10.289 31.182-24.813l30.184-130.958A203.637 203.637 0 0 0 448 310.564V179c0-40.62-35.523-71.992-75.43-66.359zm27.427 197.922c0 11.731-1.334 23.469-3.965 34.886L368.707 464h-201.92L51.591 302.303c-14.439-20.27 15.023-42.776 29.394-22.605l27.128 38.079c8.995 12.626 29.031 6.287 29.031-9.283V102c0-25.645 36.571-24.81 36.571.691V256c0 8.837 7.163 16 16 16h6.856c8.837 0 16-7.163 16-16V67c0-25.663 36.571-24.81 36.571.691V256c0 8.837 7.163 16 16 16h6.856c8.837 0 16-7.163 16-16V101.125c0-25.672 36.57-24.81 36.57.691V256c0 8.837 7.163 16 16 16h6.857c8.837 0 16-7.163 16-16v-76.309c0-26.242 36.57-25.64 36.57-.691v131.563z"
          />
        </svg>
        0
      </div>
      <div>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20,2H4C2.897,2,2,2.897,2,4v12c0,1.103,0.897,2,2,2h3v3.767L13.277,18H20c1.103,0,2-0.897,2-2V4C22,2.897,21.103,2,20,2z M20,16h-7.277L9,18.233V16H4V4h16V16z"
          />
        </svg>
         
        0
      </div>
      <div>
        <div>
          Responses
        </div>
        <div />
      </div>
      <span
        className="pointer"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="22"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M8 12l5 3V3a2 2 0 00-2-2H5a2 2 0 00-2 2v12l5-3zm-4 1.234l4-2.4 4 2.4V3a1 1 0 00-1-1H5a1 1 0 00-1 1v10.234z"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </div>
  </div>
  <article
    className="w-75"
  >
    <h1>
      
    </h1>
    <div>
      <div>
        <div>
          <img
            alt=""
            src=""
          />
        </div>
        <div>
          
           
          <button>
            Follow
          </button>
        </div>
      </div>
      <div>
        <svg
          fill="currentColor"
          height="28"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="28"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="28"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="28"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM349.3 793.7H230.6V411.9h118.7v381.8zm-59.3-434a68.8 68.8 0 1 1 68.8-68.8c-.1 38-30.9 68.8-68.8 68.8zm503.7 434H675.1V608c0-44.3-.8-101.2-61.7-101.2-61.7 0-71.2 48.2-71.2 98v188.9H423.7V411.9h113.8v52.2h1.6c15.8-30 54.5-61.7 112.3-61.7 120.2 0 142.3 79.1 142.3 181.9v209.4z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="28"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="28"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-92.4 233.5h-63.9c-50.1 0-59.8 23.8-59.8 58.8v77.1h119.6l-15.6 120.7h-104V912H539.2V602.2H434.9V481.4h104.3v-89c0-103.3 63.1-159.6 155.3-159.6 44.2 0 82.1 3.3 93.2 4.8v107.9z"
          />
        </svg>
        <div>
          <div>
            <span
              className="pointer"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="22"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 16 16"
                width="22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clipRule="evenodd"
                  d="M8 12l5 3V3a2 2 0 00-2-2H5a2 2 0 00-2 2v12l5-3zm-4 1.234l4-2.4 4 2.4V3a1 1 0 00-1-1H5a1 1 0 00-1 1v10.234z"
                  fillRule="evenodd"
                />
              </svg>
            </span>
          </div>
          <div>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M3 9.5a1.5 1.5 0 110-3 1.5 1.5 0 010 3zm5 0a1.5 1.5 0 110-3 1.5 1.5 0 010 3zm5 0a1.5 1.5 0 110-3 1.5 1.5 0 010 3z"
                fillRule="evenodd"
              />
            </svg>
          </div>
          <div>
            <li
              onClick={[Function]}
            >
              Dismiss this story
            </li>
            <li
              onClick={[Function]}
            >
              Mute this author
            </li>
            <li
              onClick={[Function]}
            >
              Mute this publication
            </li>
            <li
              onClick={[Function]}
            >
              Report this story
            </li>
            <li
              onClick={[Function]}
            >
              Block this author
            </li>
          </div>
        </div>
      </div>
    </div>
    <img
      alt=""
      className="w-100"
      onClick={[Function]}
      src=""
    />
    <div>
      <div>
        <img
          alt=""
          onClick={[Function]}
          src=""
        />
      </div>
    </div>
    <p
      className="m-4"
    >
      
    </p>
     
  </article>
</div>
`;
